infer/internal/ende/ende_test.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2022, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ende
```

Comment:
I addition to round trip tests, I wonder if we should have tests that show that modified values get the decoded modifiers reapplied to them as well?

Reply:
We exercise this functionality pretty heavily in the rest of our tests, but I agree that would be good to have. Any idea how we generate those tests? The challenge is in generating matching mutation functions at the go value and `resource.PropertyValue` level.

Reply:
If we're already exercising this pretty well in other tests; that's probably sufficient. I agree that there is no easy way to generate the matching mutations in general. I just wanted to check that we're exercising this with some example cases.